lib/puppetserver/ca/local_certificate_authority.rb

```diff
@@ -235,6 +235,13 @@ def create_intermediate_cert(root_key, root_cert)
         return nil
       end
 
+      def create_intermediate_csr
+        key = @host.create_private_key(@settings[:keylength])
+        csr = @host.create_csr(name: @settings[:ca_name], key: key)
```

Comment:
I think we want to add the `CA_EXSTENSIONS` to the CSR in this case, since we don't control the signing process. There is an example of that in here https://github.com/puppetlabs/puppet/blob/master/lib/puppet/ssl/certificate_request.rb in Puppet. Not sure exactly which bit we need, but likely something like https://github.com/puppetlabs/puppet/blob/master/lib/puppet/ssl/certificate_request.rb#L248 and https://github.com/puppetlabs/puppet/blob/master/lib/puppet/ssl/certificate_request.rb#L95.

Reply:
should be working and tested